go/install/install_unix.go

```diff
@@ -45,10 +51,19 @@ func autostartFilePath(context Context) string {
 
 // AutoInstall installs auto start on unix
 func AutoInstall(context Context, _ string, _ bool, log Log) ( /* newProc */ bool, error) {
-	// If the desktop file already exists, short circuit.
-	if _, err := os.Stat(autostartFilePath(context)); err == nil {
-		return false, nil
+	// If the desktop file already exists and has been disabled by the user, short circuit.
+	if file, err := os.Open(autostartFilePath(context)); err == nil {
+		defer file.Close()
+		// Check if the user has disabled autostart manually.
+		// If the user inserts Hidden=true, our autostart will be ignored.
+		// GNOME inserts X-GNOME-Autostart-enabled=false when startup apps are disabled via gnome-session-properties.
+		autostartDisabledRegex := regexp.MustCompile(`(?m)^Hidden=true|^X-GNOME-Autostart-enabled=false`)
```

Comment:
I can never remember precedence rules in regexes. Are we sure `|` binds last? :)